test/test_nestedtensor.py

```diff
@@ -4231,6 +4231,26 @@ def test_threshold_backward(self, device):
 
         self.assertEqual(res_dense, res_nt.values())
 
+    @onlyCUDA
+    @dtypes(torch.float32)
+    def test_record_stream(self, device, dtype):
+        nt = random_nt_from_dims(
+            [100, None, 100],
+            device=device,
+            dtype=dtype,
+            layout=torch.jagged,
+        )
+
+        expected = nt + 1
+
+        s = torch.cuda.Stream()
+        s.wait_stream(torch.cuda.default_stream(device))
+        with torch.cuda.stream(s):
+            nt.add_(1.0)
+            nt.record_stream(s)
+
+        self.assertEqual(nt, expected)
```

Comment:
You don't wait on a side stream here, so the test passes by sheer accident. Also, this doesn't actually test `record_stream` functionality, which is that memory is not reused when the tensor goes out of scope, until the stream finishes. To test this I'd do something like
```
def fn():
    x = torch.ones(100, device="cuda") # make it nested
    data_ptr = x.data_ptr() # appropriate call for nested tensor
    s = torch.cuda.Stream()
    
    with torch.cuda.stream(s):
          # do something long, including sleep
          x.record_stream(s)
    return data_ptr

data_ptr = fn()
#allocate new tensor of same size, check that its data_ptr is not test tensor data_ptr

Reply:
@ngimel I must be doing something wrong here; I can't get the test to fail without the `record_stream()` call. I'm using `torch._cuda.sleep()` to simulate doing something long and it appears to block?